core/src/main/java/io/grpc/transport/ServerTransportListener.java

```diff
@@ -48,4 +49,9 @@
    */
   ServerStreamListener streamCreated(ServerStream stream, String method,
       Metadata.Headers headers);
+
+  /**
+   * The transport completed shutting down. All resources have been released.
+   */
+  void transportTerminated();
```

Comment:
I think this would be more clear if you used similar terminology in the `ServerTransport` interface. Maybe `transportShutdown`? Or if you have `ServerTransport` extend `Closeable`, this could be `onClosed()`.